hw/ip/pwm/data/pwm.hjson

```diff
@@ -0,0 +1,201 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{ name: "pwm",
+  clock_primary: "clk_i",
+  other_clock_list: [ "clk_core_i" ],
+  reset_primary: "rst_ni",
+  other_reset_list: [ "rst_core_ni" ],
+  bus_device: "tlul",
+  bus_host: "none",
+  regwidth: "32",
+  param_list: [
+    { name: "NOutputs",
+      desc: "Number of PWM outputs",
+       type: "int",
+      default: "6",
+    }
+  ]
+  available_output_list: [
+    { name:  "pwm"
+      desc:  "pulse output"
+      width: "6"
+    }
+  ]
+  registers: [
+    { name: "REGEN",
+      desc: "Register write enable for all control registers",
+      swaccess: "rw1c",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0",
+          desc: ''' When true, all writable registers can be modified.
+                    When false, they become read-only. Defaults true, write
+                    one to clear. This can be cleared after initial
+                    configuration at boot in order to lock in the listed
+                    register settings.'''
```

Comment:
Not that it really matters, but this might be more easy to reason about if it was "REGLOCK" or similar, starting zero, rw1s. Then the thing that software writes is the thing that gets stored in the register. Alternatively, maybe it would be better rw0c?

Reply:
Hi @rswarbrick: I was just taking this from similar registers found in entropy source, or say: [alert_handler](https://github.com/lowRISC/opentitan/blob/2f6c2b019a0e56c798e8859cb673d1d237e04302/hw/ip/alert_handler/data/alert_handler.hjson#L157).   Is this being done differently in different IPs?  If so then maybe we should work to standardize them...

Reply:
Hmm, maybe. I just thought that it would be more obvious to a software author if they cleared a register by writing zero and set it by writing one. No biggie though.

Reply:
Yeah, FWIW, I agree with you. However this same register exists in many IPs with this structure.  I'd like to leave it consistent.